### ISSUE TITLE
Why tiled variable "train_ordered", "train_random", "log_alpha"?

### ISSUE THREAD

**Issue by gfdsah236** [HeddaCohenIndelman/Learning-Gumbel-Sinkhorn-Permutations-w-Pytorch#6]
Hi! Hedda
I'm confused why we must tile these variables. 
Does it help train permutation matrix? 
Is it OK if delete .repeat function?

**Comment by HeddaCohenIndelman** [comment]
Hello,
The tiling is needed if samples_per_num is greater than one, in which case learning is from multiple perturbed representations. Then, multiple samples of Gumbel noise in the dimension of the permutation representation are sampled. Each such sample is added to the permutation representation and the loss is the average over batch samples and their perturbations.
Such a learning typically generalizes better, since it allows learning from multiple examples, which are slight variations of a single representation. However, if you set samples_per_num to a constant 1, you can omit the tiling part in all affected code segments.
I hope this helps!

**Comment by gfdsah236** [comment]
Thank you for your answer.  This does help me. 
I found that the loss is higher with 'samples_per_num=1' than that with 'samples_per_num=5'.
